File: java/src/Ui/ListProductsCommand.java
package Ui;

import Products.Catalogue;

public class ListProductsCommand implements UserCommand {

	private Catalogue catalogue;

	public ListProductsCommand(Catalogue catalogue) {
		this.catalogue = catalogue;
	}

	public void run(UserInput cmd) {
		catalogue.list(System.out);							// TODO -- filter/search
	}

}

File: dotnet/CatalogueActions/ListProductsCommand.cs
namespace CatalogueActions {

using Products;
using Ui;

class ListProductsCommand : UserCommand {

	private Catalogue catalogue;

	internal ListProductsCommand(Catalogue catalogue) {
		this.catalogue = catalogue;
	}

	public void run(UserInput cmd) {
		catalogue.list();
	}

}
}